blueocean-config/src/main/java/io/jenkins/blueocean/config/BlueOceanConfigFactory.java

```diff
@@ -0,0 +1,62 @@
+package io.jenkins.blueocean.config;
+
+import javax.annotation.Nonnull;
+
+import hudson.ExtensionPoint;
+import jenkins.model.Jenkins;
+
+/**
+ * Blue Ocean configuration factory which allows different extensions to override the default behavior.
+ * 
+ */
+public abstract class BlueOceanConfigFactory implements ExtensionPoint {
+    public static final String FEATURE_PROPERTY_PREFIX = "blueocean.features.";
+    public static final String ORGANIZATIONS_ENABLED = "organizations.enabled";
+
+    private static BlueOceanConfig config;
```

Comment:
I didn't intend it like a cache, but as the loaded config, no expiration needed or any of the features a cache would give you.

If we only load this at boot, and the synchronized method is only called at that moment, the Read/Write lock would only make the code more complicated and not give you more performance.